python/lib/communication/dmod/communication/client.py

```diff
@@ -115,135 +104,277 @@ async def async_recv(self) -> str:
         """
         pass
 
+
+class SSLSecuredTransportLayerClient(TransportLayerClient, ABC):
```

Comment:
After thinking a little about it, I'd prefer not to accept an `SSLContext` directly.  I'm splitting hairs a bit, but this strictly speaking would increase the ways we could end up with an improperly/incompletely initialized context in the client.  If we did that, I would want us to somehow ensure that the `_SSLMethod` was right and that the context had right the CA certs loaded.  I think its easier to just be responsible for creating the context object.

We could accept an optional SSL directory as part of the `TransportLayerClient`'s initializer and create the context there instead of the lazy property.  Then we could still remove `SSLSecuredTransportLayerClient`.   What are your thoughts?

Reply:
I think that is a fair tradeoff, introducing a little complexity to avoid misuse. My only request would be to provide a way to pass a default ssl context.

Reply:
I've updated things to parameterize whether to use a default context.